tsfile/src/main/java/org/apache/iotdb/tsfile/encoding/encoder/TSEncodingBuilder.java

```diff
@@ -149,15 +161,18 @@ public Encoder getEncoder(TSDataType type) {
     @Override
     public void initFromProps(Map<String, String> props) {
       // set max error from initialized map or default value if not set
-      if (props == null || !props.containsKey(Encoder.MAX_POINT_NUMBER)) {
+      if (!isKeyContainedInMap(props, Encoder.MAX_POINT_NUMBER)) {
         maxPointNumber = TSFileConfig.floatPrecision;
       } else {
-        maxPointNumber = Integer.valueOf(props.get(Encoder.MAX_POINT_NUMBER));
-        if (maxPointNumber < 0) {
-          maxPointNumber = TSFileConfig.floatPrecision;
-          LOGGER
-              .warn("cannot set max point number to negative value, replaced with default value:{}",
-                  maxPointNumber);
+        for(Map.Entry<String, String> entry : props.entrySet()) {
+          if(Encoder.MAX_POINT_NUMBER.equalsIgnoreCase(entry.getKey())) {
+            maxPointNumber = Integer.valueOf(entry.getValue());
+            if (maxPointNumber < 0) {
+              maxPointNumber = TSFileConfig.floatPrecision;
+              LOGGER.warn("cannot set max point number to negative value, replaced with default value:{}",
+                        maxPointNumber);
+            }
+          }
```

Comment:
It seems that the `else` block is just enough and the `if` block is redundant.
You may set `maxPointNumber = TSFileConfig.floatPrecision` first and then go through the loop `for(Map.Entry<String, String> entry : props.entrySet()) `.